actors/core/src/main/java/cloud/orbit/actors/cluster/NodeAddress.java

```diff
@@ -33,7 +33,7 @@
 /**
  * Generic representation of a node address.
  */
-public interface NodeAddress
+public interface NodeAddress extends Comparable<NodeAddress>
 {
     UUID asUUID();
```

Comment:
Why require a node address to be a UUID?

We are running some large clusters in Marketplace and it would be super useful if we are able to make container Id and EC2 host Id part of the address. Having the actual host Id would also be helpful in this new scheme where each node can report its view of the cluster.

Reply:
That requirement is a pre-existing condition of Orbit, not something I've added here. Orbit uses that NodeAddress UUID internally in ways I don't fully understand, so I chose not to mess with it.

If you look at `RedisClusterPeer::join` (in the corresponding PR for orbit-redis-cluster here: https://github.com/ceckhardt/orbit-redis-cluster/pull/5 ), you'll see that I've added provision for an application-defined "nodeName" field to be given to the `RedisClusterTracker`, which will share that information to other nodes via `RedisClusterHeartBeat` to other machines in the cluster. That information also appears in `ClusterNodeView`.

That "node name" value originates from `@Config("orbit.actors.nodeName") Stage::nodeName`. You could easily put an EC2 host:port value there.